### PR TITLE
Issue 435

### PR DESCRIPTION
**Pull request by shivasaurus**
Connects #435 . Adds calculation modal for "Moisture in Combustion Air" in Flue Gas setup for Process Heating.

### CODE REVIEW

File: src/app/fsat/fsat-fluid/fsat-fluid.component.ts [hunk 1]
```diff
@@ -56,6 +56,8 @@ export class FsatFluidComponent implements OnInit {
     private gasDensityFormService: GasDensityFormService) { }
 
   ngOnInit() {
+    console.log("Base Gas Density: ");
+    console.log(this.baseGasDensity);
```
Comment (rmroot) [inline src/app/fsat/fsat-fluid/fsat-fluid.component.ts:60]: Remove console log

File: src/app/phast/losses/flue-gas-losses/flue-gas-losses-form-mass/flue-gas-losses-form-mass.component.html [hunk 2]
```diff
@@ -116,9 +116,9 @@
     <div class="form-group">
       <label class="small" for="{{'moistureInAirComposition'+idString}}">Moisture in Combustion Air</label>
       <div class="input-group" [ngClass]="{'indicate-different': compareMassMoistureInAirComposition(), 'error': warnings.moistureInAirCompositionWarning !== null, 'invalid': flueGasLossForm.controls.moistureInAirComposition.invalid}">
-        <input [readonly]="!baselineSelected" name="{{'moistureInAirComposition_'+lossIndex}}" type="number" step="any"
+        <input [readonly]="!baselineSelected" name="{{'moistureInAirComposition_'+lossIndex}}" type="number" step="any" 
           min="0" max="100" class="form-control" formControlName="moistureInAirComposition" id="{{'moistureInAirComposition'+idString}}"
-          onfocus="this.select();" (input)="calcExcessAir()" (focus)="focusField('moistureInAirComposition')" >
+          onfocus="this.select();" (input)="calcExcessAir()" (focus)="focusField('moistureInAirComposition')" [ngModel] = "humidityRatio | number:'1.0-2' | removeCommas" (ngModelChange) = "humidityRatio=$event">
```
Comment (rmroot) [inline src/app/phast/losses/flue-gas-losses/flue-gas-losses-form-mass/flue-gas-losses-form-mass.component.html:121]: This should just be [(ngModel)]="humidityRatio"

File: src/app/phast/losses/flue-gas-losses/flue-gas-moisture-help/flue-gas-moisture-help.component.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  selector: 'app-flue-gas-moisture-help',
+  templateUrl: './flue-gas-moisture-help.component.html',
+  styleUrls: ['./flue-gas-moisture-help.component.css']
+})
+export class FlueGasMoistureHelpComponent implements OnInit {
+  @Input()
+  currentField: string;
+  fieldSubscription: Subscription;
+  constructor(private flueGasCompareService: FlueGasCompareService) {
+    this.fieldSubscription = this.flueGasCompareService.currentField.subscribe(val => {
```
Comment (rmroot) [inline src/app/phast/losses/flue-gas-losses/flue-gas-moisture-help/flue-gas-moisture-help.component.ts:15]: should put this subscription in ngOnInit

**Review by rmroot — CHANGES_REQUESTED** [review]

File: src/app/phast/losses/flue-gas-losses/flue-gas-losses-form-mass/flue-gas-losses-form-mass.component.html (hunk 2 above)
Comment (rmroot) [reply src/app/phast/losses/flue-gas-losses/flue-gas-losses-form-mass/flue-gas-losses-form-mass.component.html:121]: If you have a reason for not doing it that way let me know

Comment (shivasaurus) [reply src/app/phast/losses/flue-gas-losses/flue-gas-losses-form-mass/flue-gas-losses-form-mass.component.html:121]: This way allows pipes to be used to format the result from the modal, but doesn't have pipes when storing data manually entered into the input (where their use wouldn't make sense).

**Comment by shivasaurus** [comment]
Removed console log and moved the subscription to ngOnInit(), replied to comment about ngModel.

**Review by rmroot — APPROVED** [review]

**Pull request merged**